src/V3Number.cpp

```diff
@@ -901,13 +897,15 @@ uint8_t V3Number::dataByte(int byte) const {
 
 bool V3Number::isEqZero() const {
     for (int i = 0; i < words(); i++) {
-        if (m_value[i] || m_valueX[i]) return false;
+        const ValueAndX v = m_value[i];
+        if (v.m_value || v.m_valueX) return false;
```

Comment:
perhaps use const ref in these cases?

Reply:
sizeof(ValueAndX) is 8 that is as same as sizeof(void*). I prefer to copy than refer for such smaller objects.
I have been thinking C++ compiler would optimize and make same binary anyway, but seems not.
Copy version is more compact on GCC-10.2.0. I'd like to leave then copied.

### Copy
```asm
00000000003d17e0 <V3Number::isEqZero() const>:
    int width() const { return m_width; }
  3d17e0:       8b 07                   mov    (%rdi),%eax
bool V3Number::isEqZero() const {
  3d17e2:       49 89 f8                mov    %rdi,%r8
    int words() const { return ((width() + 31) / 32); }
  3d17e5:       89 c2                   mov    %eax,%edx
  3d17e7:       8d 78 3e                lea    0x3e(%rax),%edi
  3d17ea:       83 c2 1f                add    $0x1f,%edx
  3d17ed:       0f 49 fa                cmovns %edx,%edi
  3d17f0:       c1 ff 05                sar    $0x5,%edi
    for (int i = 0; i < words(); i++) {
  3d17f3:       85 c0                   test   %eax,%eax
  3d17f5:       7e 41                   jle    3d1838 <V3Number::isEqZero() const+0x58>
  3d17f7:       49 c7 c1 d8 ff ff ff    mov    $0xffffffffffffffd8,%r9
  3d17fe:       49 8d 48 18             lea    0x18(%r8),%rcx
  3d1802:       31 d2                   xor    %edx,%edx
  3d1804:       4d 29 c1                sub    %r8,%r9
  3d1807:       eb 12                   jmp    3d181b <V3Number::isEqZero() const+0x3b>
  3d1809:       0f 1f 80 00 00 00 00    nopl   0x0(%rax)
  3d1810:       83 c2 01                add    $0x1,%edx
  3d1813:       48 83 c1 08             add    $0x8,%rcx
  3d1817:       39 fa                   cmp    %edi,%edx
  3d1819:       7d 1d                   jge    3d1838 <V3Number::isEqZero() const+0x58>
        return idx >= m_inlinedSize ? m_data[idx - m_inlinedSize] : m_inlined[idx];
  3d181b:       48 89 c8                mov    %rcx,%rax
  3d181e:       83 fa 01                cmp    $0x1,%edx
  3d1821:       7e 08                   jle    3d182b <V3Number::isEqZero() const+0x4b>
  3d1823:       49 8d 04 09             lea    (%r9,%rcx,1),%rax
  3d1827:       49 03 40 28             add    0x28(%r8),%rax
        if (v.m_value || v.m_valueX) return false;
  3d182b:       8b 30                   mov    (%rax),%esi
  3d182d:       0b 70 04                or     0x4(%rax),%esi
  3d1830:       74 de                   je     3d1810 <V3Number::isEqZero() const+0x30>
  3d1832:       31 c0                   xor    %eax,%eax
}
  3d1834:       c3                      retq   
  3d1835:       0f 1f 00                nopl   (%rax)
    return true;
  3d1838:       b8 01 00 00 00          mov    $0x1,%eax
  3d183d:       c3                      retq   
  3d183e:       66 90                   xchg   %ax,%ax

```

### Const Reference
```asm
00000000003d17e0 <V3Number::isEqZero() const>:
    int width() const { return m_width; }
  3d17e0:       8b 07                   mov    (%rdi),%eax
    int words() const { return ((width() + 31) / 32); }
  3d17e2:       89 c2                   mov    %eax,%edx
  3d17e4:       8d 70 3e                lea    0x3e(%rax),%esi
  3d17e7:       83 c2 1f                add    $0x1f,%edx
  3d17ea:       0f 49 f2                cmovns %edx,%esi
  3d17ed:       c1 fe 05                sar    $0x5,%esi
    for (int i = 0; i < words(); i++) {
  3d17f0:       85 c0                   test   %eax,%eax
  3d17f2:       7e 4c                   jle    3d1840 <V3Number::isEqZero() const+0x60>
  3d17f4:       49 c7 c0 d8 ff ff ff    mov    $0xffffffffffffffd8,%r8
  3d17fb:       48 8d 4f 18             lea    0x18(%rdi),%rcx
  3d17ff:       31 d2                   xor    %edx,%edx
  3d1801:       49 29 f8                sub    %rdi,%r8
  3d1804:       eb 1c                   jmp    3d1822 <V3Number::isEqZero() const+0x42>
  3d1806:       66 2e 0f 1f 84 00 00    nopw   %cs:0x0(%rax,%rax,1)
  3d180d:       00 00 00 
        if (v.m_value || v.m_valueX) return false;
  3d1810:       8b 40 04                mov    0x4(%rax),%eax
  3d1813:       85 c0                   test   %eax,%eax
  3d1815:       75 23                   jne    3d183a <V3Number::isEqZero() const+0x5a>
    for (int i = 0; i < words(); i++) {
  3d1817:       83 c2 01                add    $0x1,%edx
  3d181a:       48 83 c1 08             add    $0x8,%rcx
  3d181e:       39 f2                   cmp    %esi,%edx
  3d1820:       7d 1e                   jge    3d1840 <V3Number::isEqZero() const+0x60>
        return idx >= m_inlinedSize ? m_data[idx - m_inlinedSize] : m_inlined[idx];
  3d1822:       48 89 c8                mov    %rcx,%rax
  3d1825:       83 fa 01                cmp    $0x1,%edx
  3d1828:       7e 08                   jle    3d1832 <V3Number::isEqZero() const+0x52>
  3d182a:       49 8d 04 08             lea    (%r8,%rcx,1),%rax
  3d182e:       48 03 47 28             add    0x28(%rdi),%rax
        if (v.m_value || v.m_valueX) return false;
  3d1832:       44 8b 08                mov    (%rax),%r9d
  3d1835:       45 85 c9                test   %r9d,%r9d
  3d1838:       74 d6                   je     3d1810 <V3Number::isEqZero() const+0x30>
  3d183a:       31 c0                   xor    %eax,%eax
}
  3d183c:       c3                      retq   
  3d183d:       0f 1f 00                nopl   (%rax)
    return true;
  3d1840:       b8 01 00 00 00          mov    $0x1,%eax
  3d1845:       c3                      retq   
  3d1846:       66 2e 0f 1f 84 00 00    nopw   %cs:0x0(%rax,%rax,1)
  3d184d:       00 00 00 

```